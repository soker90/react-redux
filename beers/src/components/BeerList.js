import React, { Component } from 'react'

import BeerItem from './BeerItem'

class BeerList extends Component {

  render() {
    const {beerList, removeBeer, loading} = this.props

    return (
      <div>
        {beerList.map((beer, idx) => <BeerItem key={idx} beer={beer} loading={loading} onRemove={() => removeBeer(beer.id)} />)}
      </div>
    )
  }
}


export default BeerList
