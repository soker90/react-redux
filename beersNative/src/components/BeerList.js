import React, { Component } from 'react'

import BeerItem from './BeerItem'
import { View } from 'react-native'

class BeerList extends Component {

  render() {
    const {beerList, removeBeer, loading} = this.props

    return (
      <View>
        {beerList.map((beer, idx) => <BeerItem key={idx} beer={beer} loading={loading} onRemove={() => removeBeer(beer.id)} />)}
      </View>
    )
  }
}


export default BeerList
