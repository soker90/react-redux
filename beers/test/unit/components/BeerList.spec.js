// BeerList.spec.js
import React from 'react'
import renderer from 'react-test-renderer'

import BeerList from '../../../src/components/BeerList'
import mockData from '../mockData'

describe('BeerList component', () => {
  test('render BeerItem component without crashing', () => {
    const beerList = mockData.beers
    const loading = false
    const onRemove = jest.fn()

    const tree = renderer.create(
      <BeerList beerList={beerList} onRemove={onRemove} loading={loading}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('render BeerItem component with loading true', () => {
    const beerList = mockData.beers
    const loading = true
    const onRemove = jest.fn()

    const tree = renderer.create(
      <BeerList beerList={beerList} onRemove={onRemove} loading={loading}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

})
