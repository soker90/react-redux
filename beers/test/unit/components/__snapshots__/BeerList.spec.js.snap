exports[`BeerList component render BeerItem component with loading true 1`] = `
<div>
  <div>
    <p>
      <b>
        Amstel
      </b>
       (
      5
      %) 
      <i>
        beer_01
      </i>
    </p>
    <div>
      <button
        disabled={true}
        onClick={[Function]}>
        Loading...
      </button>
    </div>
  </div>
  <div>
    <p>
      <b>
        Mahoo
      </b>
       (
      3
      %) 
      <i>
        beer_02
      </i>
    </p>
    <div>
      <button
        disabled={true}
        onClick={[Function]}>
        Loading...
      </button>
    </div>
  </div>
  <div>
    <p>
      <b>
        Corona
      </b>
       (
      4,5
      %) 
      <i>
        beer_03
      </i>
    </p>
    <div>
      <button
        disabled={true}
        onClick={[Function]}>
        Loading...
      </button>
    </div>
  </div>
</div>
`;

exports[`BeerList component render BeerItem component without crashing 1`] = `
<div>
  <div>
    <p>
      <b>
        Amstel
      </b>
       (
      5
      %) 
      <i>
        beer_01
      </i>
    </p>
    <div>
      <button
        disabled={false}
        onClick={[Function]}>
        Remove Beer
      </button>
    </div>
  </div>
  <div>
    <p>
      <b>
        Mahoo
      </b>
       (
      3
      %) 
      <i>
        beer_02
      </i>
    </p>
    <div>
      <button
        disabled={false}
        onClick={[Function]}>
        Remove Beer
      </button>
    </div>
  </div>
  <div>
    <p>
      <b>
        Corona
      </b>
       (
      4,5
      %) 
      <i>
        beer_03
      </i>
    </p>
    <div>
      <button
        disabled={false}
        onClick={[Function]}>
        Remove Beer
      </button>
    </div>
  </div>
</div>
`;
